plugins/shuffle/round.py

```diff
@@ -199,8 +199,10 @@ def process_announcement(self):
             messages = self.inbox[phase]
             self.encryption_keys = dict()
             self.change_addresses = {}
-            for message in messages:
-                self.messages.packets.ParseFromString(messages[message])
+            for message, msg in messages.items():
+                bad = self._parse_from_string_safe(msg, 'bad')
+                if bad:
+                    continue
```

Comment:
We don't abort protocol if we receive a bad message?

Just an efficiency issue but since we don't remove the bad message, whatever parent loop is calling this will qualify every time as "full".

Reply:
Not always.  Sometimes you know "who to blame" so I tried to blame where I could (since the message has an originating key).

In some cases an abort is what makes sense...